modules/programs/yazi.nix

```
@@ -178,10 +194,29 @@ in
         }
       '';
     };
+
+    finalPackage = lib.mkOption {
+      type = types.package;
+      readOnly = true;
+      visible = false;
+      default =
+        let
+          yaziWithExtraPackages = cfg.package.override (old: {
+            extraPackages = (old.extraPackages or [ ]) ++ cfg.extraPackages;
+          });
+        in
+        if builtins.hasAttr "override" cfg.package && cfg.extraPackages != [ ] then
+          yaziWithExtraPackages
+        else
+          cfg.package;
```

I think the use of `lib` is generally preferred over `builtins`, since the latter is tied to the Nix version (which might change the behaviour), but aside from that, we can shorten this to:

```suggestion
        if cfg.package ? override && cfg.extraPackages != [ ] then yaziWithExtraPackages else cfg.package;
```

This said, isn't the `override` attribute always available?

```
nix-repl> pkgs.yazi ? override
true
```

---

Technically, it's not always available... more likely than not it will be. But, just wanted to make it follows the other pattern and be safer.